src/wallet/test/wallet_tests.cpp

```diff
@@ -379,4 +379,34 @@ BOOST_FIXTURE_TEST_CASE(generate_with_incorrect_privkey, TestingSetup)
 
 }
 
+BOOST_FIXTURE_TEST_CASE(getTokenBalance, TestChainSetup)
+{
+    CBlockIndex* const nullBlock = nullptr;
+    CBlockIndex* oldTip = chainActive.Tip();
+    GetBlockFileInfo(oldTip->GetBlockPos().nFile)->nSize = MAX_BLOCKFILE_SIZE;
+    // CreateAndProcessBlock({}, GetScriptForRawPubKey(coinbaseKey.GetPubKey()));
+    CBlockIndex* newTip = chainActive.Tip();
+
+    // CP2PKH(Colored P2PKH)：
+    // <COLOR identifier> OP_COLOR OP_DUP OP_HASH160 <H(pubkey)> OP_EQUALVERIFY OP_CHECKSIG
+    CScript CP2PKHScriptPubKey = CScript() << ParseHex("011863143c14c5166804bd19203356da136c985678cd4d27a1b8c6329604903262") << OP_COLOR << OP_DUP << OP_HASH160 << ParseHex("1018853670f9f3b0582c5b9ee8ce93764ac32b93") << OP_EQUALVERIFY << OP_CHECKSIG;
+
+    // CP2SH(Colored P2SH)：
+    // <COLOR identifier> OP_COLOR OP_HASH160 <H(redeem script)> OP_EQUAL
+    CScript CP2SHScriptPubKey = CScript() << ParseHex("011863143c14c5166804bd19203356da136c985678cd4d27a1b8c6329604903262") << OP_COLOR << OP_HASH160 << ParseHex("da8a647bba351bbae4cee0089d373c97ec240580") << OP_EQUAL;
+
+    {
+        CWallet wallet("dummy", WalletDatabase::CreateDummy());
+        AddKey(wallet, coinbaseKey);
+        WalletRescanReserver reserver(&wallet);
+        reserver.reserve();
+        CreateAndProcessBlock({}, CP2PKHScriptPubKey);
+
+        std::vector<CBalance> walletBalance = wallet.GetAvailableTokenBalance();
+        // printf("wallet size after 1: %u\n", walletBalance.size());
+
```

Comment:
Isn't it necessary to check if the result of `wallet.GetAvailableTokenBalance()` is what you expected?

Reply:
ya this one i havent completed yet

Reply:
Do you add a commit to this pull request to complete it?

Reply:
i will add a commit later

Reply:
azuchi san 
i temporary commented this unit test 
because found out there is one dependency not yet make changes
GetScriptForDestination might affect alot of places
so i will create another branch working on that